warehouse/query-core/src/main/java/datawave/core/iterators/TermFrequencyOffsetIterator.java

```diff
@@ -0,0 +1,164 @@
+package datawave.core.iterators;
+
+import org.apache.accumulo.core.data.ArrayByteSequence;
+import org.apache.accumulo.core.data.ByteSequence;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Range;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.iterators.IteratorEnvironment;
+import org.apache.accumulo.core.iterators.SortedKeyValueIterator;
+import org.apache.hadoop.io.Text;
+import org.apache.log4j.Logger;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+import java.util.SortedSet;
+import java.util.TreeSet;
+
+/**
+ * Alternate implementation of the {@link TermFrequencyIterator} that operates on precomputed column qualifiers.
+ *
+ * Term Frequency keys take the form {shard:tf:datatype\0uid\0value\0field}.
+ *
+ * This iterator fetches offsets for known datatype, uid, value, and field combinations. It does not support fetching offsets for a partial column qualifier.
+ */
+public class TermFrequencyOffsetIterator implements SortedKeyValueIterator<Key,Value> {
+    
+    private static final Logger log = Logger.getLogger(TermFrequencyOffsetIterator.class);
+    
+    private Key tk = null;
+    private Value tv = null;
+    
+    // A set of full column qualifiers that describe the offset's exact location (datatype\0uid\0value\0FIELD)
+    private final SortedSet<Text> searchSpace;
```

Comment:
I am wondering why we need this iterator.  This is essentially equivalent to supplying a set of ranges with the underlying source that include only the searchSpace values.  If we truly need this as as separate iterator, then I would point you to consider extending accumulo's SeekingFilter instead.

Reply:
I added a TermFrequencyFilter offered up as an alternative implementation

Reply:
Great work -- I removed the TermFrequencyOffsetIter in the latest commit in favor of the TermOffsetFilter